crates/stages/stages/src/stages/sender_recovery.rs

```diff
@@ -266,6 +295,10 @@ enum SenderRecoveryStageError {
     #[error(transparent)]
     FailedRecovery(#[from] FailedSenderRecoveryError),
 
+    /// A recovering batch job failed recovery
+    #[error("failed to recover a batch")]
+    FailedBatchRecovery,
```

Comment:
let's clarify that it's about mismatch of entities number

Reply:
this can also use GotExpected and include the mismatch in the error